AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java

```diff
@@ -2164,12 +2169,33 @@ public static void clearUserContent() {
     VarCache.clearUserContent();
   }
 
-  private static void parseSdkCounters(JSONObject response) {
+  @VisibleForTesting
+  public static HashSet<String> parseSdkCounters(JSONObject response) {
     JSONArray enabledCounters = response.optJSONArray(
             Constants.Keys.ENABLED_COUNTERS);
+    HashSet<String> counterSet = new HashSet<>();
     if (enabledCounters != null) {
-      HashSet counterSet = new HashSet<>(Arrays.asList(enabledCounters));
-      CountAggregator.INSTANCE.setEnabledCounters(counterSet);
+      counterSet.addAll(toList(enabledCounters));
     }
+    return counterSet;
   }
+
+  @VisibleForTesting
+  public static HashSet<String> parseFeatureFlags(JSONObject response) {
+    JSONArray enabledFeatureFlags = response.optJSONArray(
+            Constants.Keys.ENABLED_FEATURE_FLAGS);
+    HashSet<String> featureFlagSet = new HashSet<>();
+    if (enabledFeatureFlags != null) {
+      featureFlagSet.addAll(toList(enabledFeatureFlags));
+    }
+    return featureFlagSet;
+  }
+
+  private static List<String> toList(JSONArray array) {
```

Comment:
Hey, why don't you just use 
```
private static HashSet<String> toHashSet(JSONArray array) {
      HashSet<String> yourset = new HashSet<>();
      for (int i = 0; i < array.length(); i++) {
          yourset.add(array.optString(i));
      }
      return yourset;
}
```
The way converting from `JSONArray` => `List` => add all to `HashSet` is taking time and you can do it more easier.